In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats

In [2]:
data = pd.read_csv('Week_8_Q&A_dataset - Sheet1.csv')

In [3]:
data.head()

Gender Test preparation  Total Marks
0    male             none           14
1  female             none           28
2  female             none           18
3  female             none           48
4  female             none           21

**1. Descriptive Analysis in the dataset.**

In [24]:
data.isnull().sum()

Gender              0
Test preparation    0
Total Marks         0
dtype: int64

Hence there is no missing data.

In [4]:
data.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

A total of 28 student's data were taken. Out of which maximum marks scored was 50, minimum was 12, average score being 32.32 and a standard deviation of 12.45

In [46]:
data.groupby('Test preparation').count()

Gender  Total Marks
Test preparation                     
completed              7            7
none                  21           21

In [47]:
data.groupby('Gender').count()

Test preparation  Total Marks
Gender                               
female                16           16
male                  12           12

Out of 28 students, 7 completed test preparation and 21 did not. Also, there were 16 females and 12 male students.

________________________________________________________________________________

**2.Check whether female and male students scored the same marks**

In [25]:
f=data[data['Gender']=='female']
m=data[data['Gender']=='male']
fc=f['Total Marks']
mc=m['Total Marks']

In [35]:
f

Gender Test preparation  Total Marks
1   female             none           28
2   female             none           18
3   female             none           48
4   female             none           21
5   female        completed           40
7   female             none           12
14  female             none           41
15  female             none           13
16  female        completed           48
17  female             none           45
22  female             none           41
23  female             none           27
24  female             none           27
25  female             none           38
26  female        completed           50
27  female        completed           43

In [29]:
if ((fc.sum())==(mc.sum())):
    print('Both scored same',fc.sum(),'Male Score:',mc.sum())
elif((fc.sum())>(mc.sum())):
    print('Females scored more marks!\nFemale Score:',fc.sum(),'Male Score:',mc.sum())
else:
     print('Males scored more marks',fc.sum(),'Male Score:',mc.sum())

Females scored more marks!
Female Score: 540 Male Score: 365


________________________________________________________________________________

**3.Whether test preparation helps the students?**

2sample t-test: n=28(n<30)

Null Hypothesis: Test Preparation helps the students.

Alternate Hypothesis: Test Preparation dose'nt help the students

Assuming Significance level: 0.05 i.e 5%

In [21]:
print(np.var(fc), np.var(mc))

155.1875 135.74305555555554


In [31]:
tvalue, pvalue = stats.ttest_ind(fc,mc)
stats.ttest_ind(fc,mc)

Ttest_indResult(statistic=0.6940885268848668, pvalue=0.4937822317960763)

In [32]:
crit_val=stats.t.ppf(1-0.05,28)
crit_val

1.701130934265931

In [33]:
if (tvalue>crit_val and pvalue < 0.05):
    print('Reject Null Hypothesis. Hence test preparation dosent help the students')
else:
    print('Reject Alternate Hypothesis. Hence test preparation helps the students')

Reject Alternate Hypothesis. Hence there is increase in sales after stepping into digital marketing


pvalue 0.4937822317960763 is greater than the significance level 0.05, thereby we fail to reject the the null hypothesis. Hence it can be assumed that the test preparation helps the students.
____________________________________________________________________________________________________